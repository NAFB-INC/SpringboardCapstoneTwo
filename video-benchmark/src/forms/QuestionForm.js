import React, { useState } from "react";
import VideoAPI from "../api/VideoAPI";
import useFields from "../hooks/useFields";

function QuestionForm({presID,questions,setQuestions,setRevealQBox,setFeedback,fetchQuestions}) {
    const [errors,setErrors] = useState([]);
    const [formData, handleChange, resetForm] = useFields({
        question: ''
    })

    const handleSubmit = e => {
        e.preventDefault();
        if(!formData.question){
            setErrors(["Text is required for a question!"])
        }
        if(Object.values(questions)){
            for(let val of Object.values(questions)){
                if(formData.question === val){
                    setFeedback("Your question has already been asked, and has been answered, or will be answered soon!");
                    setRevealQBox(false);
                }
            }
        }
        if(errors.length < 1){
            VideoAPI.addQuestion(presID,formData.question);
            fetchQuestions();
            setFeedback("Your question has been added to the que!");
            setRevealQBox(false);
        }
        resetForm();
    }

    return (
        <div className="QuestionForm">
            <form onSubmit={handleSubmit}>
                    <label htmlFor="question">Question: </label>
                    <input name="question" type="text" placeholder="What did you mean by ___?" value={formData.question} onChange={handleChange}></input>
                    <div>
                        {errors.map((e)=>(
                            <li className="error" key={`error${e}`}>{e}</li>
                        ))}
                    </div>
                    <button id="submit-q" className="present-choice-btn-sm">Submit</button>
            </form>
        </div>
    );
}

export default QuestionForm;