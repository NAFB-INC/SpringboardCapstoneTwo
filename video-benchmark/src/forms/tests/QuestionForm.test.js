import React from "react";
import { render } from '@testing-library/react';
import QuestionForm from '../QuestionForm';

it("renders without crashing", function() {
  render(<QuestionForm />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<QuestionForm />);
  expect(asFragment()).toMatchSnapshot();
});