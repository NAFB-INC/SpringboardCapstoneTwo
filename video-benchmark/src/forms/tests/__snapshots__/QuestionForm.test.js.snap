// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="QuestionForm"
  >
    <form>
      <label
        for="question"
      >
        Question: 
      </label>
      <input
        name="question"
        placeholder="What did you mean by ___?"
        type="text"
        value=""
      />
      <div />
      <button
        class="present-choice-btn-sm"
        id="submit-q"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
